source/commands/nick.cpp

```diff
@@ -1,9 +1,38 @@
 #include "commands.hpp"
 
+bool checkNick(std::string nick)
```

Comment:
The whole Nickname should be checked. RFC1459 says: `<letter> { <letter> | <number> | <special> }`. We might want to think about adding functions like `is_letter`, `is_number`, `is_nonwhite`, etc. to a common place, so we have to write them only once.